tests/integration/kubernetes/kata-deploy-test-usig-nfd.bats

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env bats
```

Comment:
I think kata-deploy should be made on its own suite, e.g. `tests/functional/kata-deploy`, otherwise `tests/integration/kubernetes` will be bloated over time. Putting the hat of downstream projects (CoCo operator, peer pods, for example), we don't need to run kata-deploy tests along-side kubernetes' s.

Reply:
We actually have to run those kata-deploy tests as part of the CoCo operator, as the kata-deploy payload is exactly what the operator consumes.

Reply:
I went back and forth on this, Wainer.

I'm not sure if we want to have this as part of kubernetes, but we do need the majority of the infra that's now part of the kubernetes tests.  Changing this to its own directory will require some reasonable refactoring / reuse, which I can do, but keep in mind that there may be code duplication.